examples/trusted_information_retrieval/module/rust/src/lib.rs

```diff
@@ -85,53 +134,86 @@ impl TrustedInformationRetrieval for TrustedInformationRetrievalNode {
     ) -> grpc::Result<ListPointsOfInterestResponse> {
         debug!("Received request: {:?}", request);
         let location = request.location.ok_or_else(|| {
-            let err = "Location is not specified".to_string();
-            error!("Bad request: {:?}", err);
-            grpc::build_status(grpc::Code::InvalidArgument, &err)
+            let error = "Location is not specified".to_string();
+            error!("Bad request: {:?}", error);
+            grpc::build_status(grpc::Code::InvalidArgument, &error)
         })?;
-        let nearest_point = self.points_of_interest.iter().fold(
-            (None, f32::MAX),
-            |(closest_point, min_distance), point| {
-                let point_location = point.location.clone().expect("Non-existing location");
-                let distance = Self::distance(location.clone(), point_location);
-                if distance < min_distance {
-                    (Some(point.clone()), distance)
-                } else {
-                    (closest_point, min_distance)
-                }
-            },
-        );
-        match nearest_point.0 {
-            Some(point) => {
+
+        // Request entries from the database until the last entry page is reached.
+        let mut current_offset = 0;
+        let mut nearest_point = None;
+        let mut last = false;
+        while !last {
+            let (points_of_interest, last_page) = self
+                .list_database_entries(current_offset, DATABASE_PAGE_SIZE)
+                .map_err(|error| {
+                    let status = grpc::build_status(grpc::Code::Unavailable, "");
+                    error!("Couldn't get database entries: {:?}, {:?}", error, status);
+                    status
+                })?;
+
+            // Find the nearest point of interest.
+            nearest_point = points_of_interest
+                .iter()
+                .fold(
+                    (nearest_point, f32::MAX),
```

Comment:
nearest_point vs closest_point seems a very confusing terminology, if the intention is just not to use the same name twice